independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java

```diff
@@ -105,7 +107,7 @@ public static int run(Path projectDir, Wrapper wrapper) {
     private static String getMavenSettingsArg() {
         final String mavenSettings = System.getProperty("maven.settings");
         if (mavenSettings != null) {
-            return mavenSettings;
+            return Files.exists(Paths.get(mavenSettings)) ? mavenSettings : null;
         }
         return BootstrapMavenOptions.newInstance().getOptionValue(BootstrapMavenOptions.ALTERNATE_USER_SETTINGS);
```

Comment:
BTW shouldn't we make this result an absolute path?

Reply:
It's worth making sure the ALTERNATE_USER_SETTINGS is an absolute path.